src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2433 @@
+    }
+
+    [Fact]
+    public void SignatureUsage_07()
+    {
+        var source = """
+            file class C
+            {
+            }
+
+            class D
+            {
+                public static D operator +(D d, C c) => d; // 1
+                public static C operator -(D d1, D d2) => new C(); // 2
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (7,30): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public static D operator +(D d, C c) => d; // 1
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "+").WithArguments("C", "D").WithLocation(7, 30),
+            // (8,30): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public static C operator -(D d1, D d2) => new C(); // 2
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "-").WithArguments("C", "D").WithLocation(8, 30));
+    }
+
+    [Fact]
+    public void SignatureUsage_08()
+    {
+        var source = """
+            file class C
+            {
+            }
+
+            class D
+            {
+                public D(C c) { } // 1
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (7,12): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public D(C c) { } // 1
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "D").WithArguments("C", "D").WithLocation(7, 12));
+    }
+
+    [Fact]
+    public void SignatureUsage_09()
+    {
+        var source = """
+            file class C
+            {
+            }
+
+            class D
+            {
+                public C M(C c1, C c2) => c1; // 1, 2, 3
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (7,14): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public C M(C c1, C c2) => c1; // 1, 2, 3
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M").WithArguments("C", "D").WithLocation(7, 14),
+            // (7,14): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public C M(C c1, C c2) => c1; // 1, 2, 3
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M").WithArguments("C", "D").WithLocation(7, 14),
+            // (7,14): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+            //     public C M(C c1, C c2) => c1; // 1, 2, 3
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M").WithArguments("C", "D").WithLocation(7, 14));
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

It looks like the code already handles this but consider testing type arguments, arrays, pointers in signatures.
```csharp
file struct S { }
class A<T> { }
class B
{
    public static void F1(A<(S, int)> a) { }
    public unsafe static void F2(S*[] a) { }
}
```

Reply:
Have added some of these cases to `TypeArguments_01`